### ISSUE TITLE
Any trailer won't unload anything / any trailer

### ISSUE THREAD

**Issue by Sam-D03** [loki79uk/FS25_UniversalAutoload#168]
Hey! Probably a mod conflict, but thought I would raise the ticket anyway.

All autoload functionality is working correctly, apart from that nothing unloads. I've tried changing the key binding to other things and this doesn't help.

Debugger looks like it knows where to unload, it just doesn't trigger the actual unload!

![Screenshot 2024-12-19 233704](https://github.com/user-attachments/assets/33b651f3-8b85-4659-8e6c-cb1d6648ca0d)

Thanks for the awesome mod though!

**Comment by loki79uk** [comment]
This sounds like a key binding issue as you say.  Try removing the UAL key bindings from the bindings XML and restart, but also feel free to attach your log file and I'll check nothing weird is happening.

**Comment by loki79uk** [comment]
Did you manage to get it working again?

**Comment by Sheoji** [comment]
![untitled](https://github.com/user-attachments/assets/79da0207-4881-43ad-a963-2b6b59650b6b)
Can confirm, the problem with trailers persist. I can unload logs, but cant load them.
I checked unload prompt with loading manually.

**Comment by loki79uk** [comment]
Would it be helpful to add some debugging prints like this?
![image](https://github.com/user-attachments/assets/e8386278-8b10-40d2-9bf5-ace68466c64a)

**Comment by Sheoji** [comment]
![image](https://github.com/user-attachments/assets/f60116f2-7426-443a-ba09-1ce4c7a737c9)
![image](https://github.com/user-attachments/assets/79f95559-4984-4769-8066-ba0cc3b17659)
![image](https://github.com/user-attachments/assets/e129e29c-2ea1-4167-bc3e-f4fb8d73b05a)
Bought a bigger boi. Same thing.

**Comment by loki79uk** [comment]
Can you do a screenshot of the bigger zone you made?  That looks exactly the same as before.  Or look in your mod settings file and tell me the length for that trailer.  It should look something like:
![image](https://github.com/user-attachments/assets/08bb682c-cacd-45cc-a79f-adff89dc14f0)

**Comment by loki79uk** [comment]
> P.S. Sorry I messed up please apply my posts to issue #171.

Okay, I just realised you are not OP and you hijacked this bug with your own issue.  It would still be helpful for you to tell me the value you have for length in the settings file.

**Comment by Sheoji** [comment]
Sorry for the confusion again, I posted in the wrong tab and will be more attentive from now on.
Log. Please note this and stock settings done nothing.
<loadingArea offset="0.015 1.053 -1.184" width="2.064340" length="4.934532" height="2.064995"/>

**Comment by loki79uk** [comment]
> Sorry for the confusion again, I posted in the wrong tab and will be more attentive from now on. Log. Please note this and stock settings done nothing. loadingArea offset="0.015 1.053 -1.184" width="2.064340" length="4.934532" height="2.064995"

Okay, so if you try to load 5m or above logs they won't load unless you resize over 5m.  Did you try cutting the logs shorter?

**Comment by Sheoji** [comment]
I will clarify:
Trailer wont load 5m log under any of the tested conditions. The same 5m truck works as intended.
I even leased a new one, and set the ridiculous dimensions.
![image](https://github.com/user-attachments/assets/096f2594-cafd-4dd5-b92e-43da8582432d)
![image](https://github.com/user-attachments/assets/f896b582-1c51-41d7-b2ca-66a48c0036f3)
I cant read code well, so i'm screening both trucks.
![image](https://github.com/user-attachments/assets/df679a2b-e58f-4b5f-bc77-88c650157750)

**Comment by loki79uk** [comment]
Are you playing in multiplayer with an old version (before 0.3.0.0)?  Are you restarting the server?  If SP are you using the same trailer that you purchased in the shop?

**Comment by Sheoji** [comment]
Updated ASAP. Tested SP only so far. All vehicles are purchased or leased on 0.3.0.
In short, I updated and done some testing on the fresh save.

**Comment by loki79uk** [comment]
> Updated ASAP. Tested SP only so far. All vehicles are purchased or leased on 0.3.0. In short, I updated and done some testing on the fresh save. I will do a retest with clean enviroment in a bit.

Okay - it is the lease button - it seems that has been broken since v0.3

It doesn't behave the same as buy for some reason..

```
2024-12-20 22:12 onYesNoLease: nil
2024-12-20 22:12 onYesNoBuy: true
```

**Comment by Sheoji** [comment]
Checked on full buy, Still cant load on trailer.
Fresh test results (all clean, new save, only autoloader):
All vehicles are bought and dimensions extended.
FH16 RUL-HKR Works as intended with no issues noted. Loads both crates and logs.
RUH327 (trailer-attachment that sold separetly):
Cant load crates, but CAN unload them.
ACTUALLY WORKS WITH LOGS NOW! (I'm investigating the cause of this ATM)
![image](https://github.com/user-attachments/assets/7a61ddbc-95b2-47d9-89f1-769faec74a03)
![image](https://github.com/user-attachments/assets/70bfa6c9-e6fb-417c-8df8-7fc1c5e3032c)
![image](https://github.com/user-attachments/assets/a1be53c0-fed6-4975-b478-9aa20de01eba)

P.S. Please let me know if you want me to post in issue 171 from now on. I really messed up on this one and this is bugging me more that issue now.

**Comment by loki79uk** [comment]
> UPD: Just spotted the update, please stand by, info in old.

If you want to just message me on Discord it is WAY faster than using GitHub: loki_79#2864

**Comment by Sheoji** [comment]
Sorry, Discord messed up PM's cant message you. Personal advice: While I was helping with VA in Anomaly mod, I created a dummy Discord server for contact and file sharing. You can use this instead personal ID.
Leased gear in question seem to work now.
My guess is that trailer is not compatible for boxes. Still can unload them if needed.
Logs are fixed.
My mod list is minimal, but still have no effect.
Consider my issue resolved and THANK YOU for you patience. <3

**Comment by loki79uk** [comment]
> Sorry, Discord messed up PM's cant message you. Personal advice: While I was helping with VA in Anomaly mod, I created a dummy Discord server for contact and file sharing. You can use this instead personal ID. Leased gear in question seem to work now. My guess is that trailer is not compatible for boxes. Still can unload them if needed. Logs are fixed. My mod list is minimal, but still have no effect. Consider my issue resolved and THANK YOU for you patience. <3

Yes, you are correct - log trailers shouldn't load pallets because most of them do not have a flat base and the pallets fall through the gaps! :D

**Comment by loki79uk** [comment]
> Hey! Probably a mod conflict, but thought I would raise the ticket anyway.
> 
> All autoload functionality is working correctly, apart from that nothing unloads. I've tried changing the key binding to other things and this doesn't help.
> 
> Debugger looks like it knows where to unload, it just doesn't trigger the actual unload!

Have you managed to resolve this?  Despite the thread hijack, you have not actually replied at all since the original post?

**Comment by Sam-D03** [comment]
Sorry for the radio silence! Had a chance to do more testing tonight. Started a new save game with zero other mods running, and still no unload at all. Tried multiple vehicles and trailers and all have the same effect, no unloading :( Super weird! Fresh Log file attached.

[log.txt](https://github.com/user-attachments/files/18223943/log.txt)

**Comment by Sam-D03** [comment]
Ok its definitely something to do with control binding. I added an additional joystick control binding to "unload" and suddenly the  tooltip option to unload became available and unloading worked again! Assigned a different joystick button, no more unloading. Then assigned again and unloading worked again. So it seems related to the control system itself.

**Comment by loki79uk** [comment]
> Ok its definitely something to do with control binding. I added an additional joystick control binding to "unload" and suddenly the tooltip option to unload became available and unloading worked again! Assigned a different joystick button, no more unloading. Then assigned again and unloading worked again. So it seems related to the control system itself.

That is interesting!  I know there are issues with bindings just stopping working, but I have never heard of that as a workaround.  I'll remember that for future debugging!

**Comment by loki79uk** [comment]
Closing as don't think this is anything I can change with UAL, it is just how the GIANTS keybinding system is designed.
